src/content_handlers/apt_handler/src/apt_parser.cpp

```diff
@@ -88,6 +92,15 @@ std::unique_ptr<AptContent> GetAptContentFromRootValue(JSON_Value* rootValue)
                     ADUC_ERC_APT_HANDLER_INVALID_PACKAGE_DATA);
             }
 
+            // Are we installing deviceupdate-agent package?
+            // Currently, we are assuming deviceupdate-agent* is a Device Update Agent package.
+            if (!aptContent->AgentRestartRequired && (name.find_first_of("deviceupdate-agent") == 0))
```

Comment:
>deviceupdate-agent [](start = 74, length = 18)

this should probably be defined at the top of this file, with a comment explaining that it's used to match if we need to reboot.